ts/session/apis/session_backend_server.ts

```diff
@@ -0,0 +1,227 @@
+import AbortController from 'abort-controller';
+import { z, ZodError, ZodSchema } from 'zod';
+import { BlindingActions } from '../../webworker/workers/browser/libsession_worker_interface';
+import { isOnionV4JSONSnodeResponse, OnionSending } from '../onions/onionSend';
+import { fromUInt8ArrayToBase64 } from '../utils/String';
+import { NetworkTime } from '../../util/NetworkTime';
+import {
+  batchGlobalIsSuccess,
+  parseBatchGlobalStatusCode,
+} from './open_group_api/sogsv3/sogsV3BatchPoll';
+import { UserUtils } from '../utils';
+import { timeoutWithAbort } from '../utils/Promise';
+import { DURATION } from '../constants';
+
+export type SessionServerConfigType = {
+  name: string;
+  url: `${'http' | 'https'}://${string}`;
+  edPkHex: string;
+  xPkHex: string;
+  timeoutMs: number;
+};
+
+type WithZodSchemaValidation<S = ZodSchema> = {
+  withZodSchema: S;
+};
+
+type SessionBackendServerMakeRequestParams = {
+  path: `/${string}`;
+  method: 'GET' | 'POST';
+  bodyGetter?: () => Promise<string | null>;
+  blindSignRequest?: boolean;
+};
+
+type BlindSignedHeaders = {
+  'X-FS-Pubkey': string;
+  'X-FS-Timestamp': string;
+  'X-FS-Signature': string;
+};
+
+export const SessionBackendBaseResponseSchema = z.object({
+  status_code: z.number(),
+  t: z.number(),
+});
+
+export type SessionBackendBaseResponseSchema = z.infer<typeof SessionBackendBaseResponseSchema>;
+
+export type SessionBackendServerApiResponse = SessionBackendBaseResponseSchema &
+  Record<string, unknown>;
+
+export default class SessionBackendServerApi {
+  readonly server: SessionServerConfigType;
+
+  constructor(server: Omit<SessionServerConfigType, 'timeoutMs'>, timeoutMsOverride?: number) {
+    this.server = {
+      ...server,
+      timeoutMs: timeoutMsOverride || 10 * DURATION.SECONDS,
+    };
+  }
+
+  private static getGenericErrorResponse() {
+    return {
+      status_code: 500,
+      t: NetworkTime.nowSeconds(),
+    };
+  }
+
+  public logInfo(msg: string, path?: string) {
+    window.log.info(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}${msg}`);
+  }
+  public logError(msg: string, path?: string) {
+    window.log.error(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}${msg}`);
+  }
+  public logZodError(zodError: ZodError, path?: string) {
+    window.log.error(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}`, zodError);
+  }
+
+  // NOTE: returns null if the ed25519 key pair is not found
+  private async getBlindSignedHeaders({
+    method,
+    path,
+    body,
+  }: {
+    method: string;
```

Comment:
Create a type and reuse it here for method type from l29?
`  method: 'GET' | 'POST';`